### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by mwenyoa**
#### Interface
- Created a class `Nameable`.
   - Implemented a method called `correct_name` that will raise a `NotImplementedError`.

#### Turned my Person class to Nameable
- Made sure that my Person class inherits from Nameable
- Made sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepared base Decorator
- Made sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Created a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Created a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  #   correct name method
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+#  capitalize name decorator
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (FrederickMih) [inline decorator.rb]: - [ ] Hi @mwenyoa, I think `upcase()` method will convert all the strings to uppercase. We are required to capitalize the first letter of the given name. To achieve this, you can use the `capitalize()` method. That's `@nameable.correct_name.capitalize`

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @mwenyoa  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




# To Highlights

- GitHub workflow has been followed :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- Linter checks are Ok :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (mwenyoa) [reply decorator.rb]: Hey @FrederickMih Thanks for that observation and helpful feedback. let me correct it right away

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @mwenyoa,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**